soc/xtensa/nxp_adsp/imx8/include/soc/memory.h

```diff
@@ -0,0 +1,251 @@
+
+#define RAM_SIZE \
+	(SRAM_SIZE - SRAM_RESERVE_HEADER_SPACE - VECTOR_TBL_SIZE)
+
+/* Location for the intList section which is later used to construct the
+ * Interrupt Descriptor Table (IDT). This is a bogus address as this
+ * section will be stripped off in the final image.
+ */
+#define IDT_BASE	(RAM_BASE + RAM_SIZE)
+
+/* size of the Interrupt Descriptor Table (IDT) */
+#define IDT_SIZE	0x2000
+
+/* physical DSP addresses */
+#define IRAM_BASE	0x596f8000
+#define IRAM_SIZE	0x800
+
+#define DRAM0_BASE	0x596e8000
+#define DRAM0_SIZE	0x8000
+
+#define DRAM1_BASE	0x596f0000
+#define DRAM1_SIZE	0x8000
+
+#define SDRAM0_BASE	0x92400000
+#define SDRAM0_SIZE	0x800000
+
+#define SDRAM1_BASE	0x92C00000
+#define SDRAM1_SIZE	0x800000
+
+#define XSHAL_MU13_SIDEB_BYPASS_PADDR 0x5D310000
+#define MU_BASE		XSHAL_MU13_SIDEB_BYPASS_PADDR
+
+#define EDMA0_BASE	0x59200000
+#define EDMA0_SIZE	0x10000
+
+#define ESAI_BASE	0x59010000
+#define ESAI_SIZE	0x00010000
+
+#define SAI_1_BASE	0x59050000
+#define SAI_1_SIZE	0x00010000
+
+#define UUID_ENTRY_ELF_BASE	0x1FFFA000
+#define UUID_ENTRY_ELF_SIZE	0x6000
+
+#define LOG_ENTRY_ELF_BASE	0x20000000
+#define LOG_ENTRY_ELF_SIZE	0x2000000
+
+#define EXT_MANIFEST_ELF_BASE	(LOG_ENTRY_ELF_BASE + LOG_ENTRY_ELF_SIZE)
+#define EXT_MANIFEST_ELF_SIZE	0x2000000
+
+/*
+ * The Heap and Stack on i.MX8 are organized like this :-
+ *
+ * +--------------------------------------------------------------------------+
+ * | Offset              | Region         |  Size                             |
+ * +---------------------+----------------+-----------------------------------+
+ * | SDRAM_BASE          | RO Data        |  SOF_DATA_SIZE                    |
+ * |                     | Data           |                                   |
+ * |                     | BSS            |                                   |
+ * +---------------------+----------------+-----------------------------------+
+ * | HEAP_SYSTEM_BASE    | System Heap    |  HEAP_SYSTEM_SIZE                 |
+ * +---------------------+----------------+-----------------------------------+
+ * | HEAP_RUNTIME_BASE   | Runtime Heap   |  HEAP_RUNTIME_SIZE                |
+ * +---------------------+----------------+-----------------------------------+
+ * | HEAP_BUFFER_BASE    | Module Buffers |  HEAP_BUFFER_SIZE                 |
+ * +---------------------+----------------+-----------------------------------+
+ * | SOF_STACK_END       | Stack          |  SOF_STACK_SIZE                   |
+ * +---------------------+----------------+-----------------------------------+
+ * | SOF_STACK_BASE      |                |                                   |
+ * +---------------------+----------------+-----------------------------------+
+ */
+
+#define SRAM_OUTBOX_BASE	SDRAM1_BASE
+#define SRAM_OUTBOX_SIZE	0x1000
+#define SRAM_OUTBOX_OFFSET	0
+
+#define SRAM_INBOX_BASE		(SRAM_OUTBOX_BASE + SRAM_OUTBOX_SIZE)
+#define SRAM_INBOX_SIZE		0x1000
+#define SRAM_INBOX_OFFSET	SRAM_OUTBOX_SIZE
```

Comment:
More stuff copied over from SOF/intel_adsp?  The inbox/outbox stuff are specific to the communication with the host x86 CPUs.  Surely they aren't relevant to you?

Reply:
We also have this on IMX for SOF with XTOS -  see https://github.com/thesofproject/sof/blob/main/src/platform/imx8/include/platform/lib/memory.h#L83